Mathlib/Topology/Instances/AddCircle.lean

```diff
@@ -267,6 +268,36 @@ theorem continuousAt_equivIoc : ContinuousAt (equivIoc p a) x := by
   exact (continuousAt_toIocMod hp.out a hx).codRestrict _
 #align add_circle.continuous_at_equiv_Ioc AddCircle.continuousAt_equivIoc
 
+/-- The quotient map `𝕜 → AddCircle p` as a partial homeomorphism. -/
+def partialHomeomorphCoe [DiscreteTopology (zmultiples p)] : PartialHomeomorph 𝕜 (AddCircle p) :=
+  { toFun := (↑)
+    invFun := fun x ↦ equivIco p a x
+    source := Ioo a (a + p)
+    target := {↑a}ᶜ
+    map_source' := by
+      intro x hx hx'
+      exact hx.1.ne' ((coe_eq_coe_iff_of_mem_Ico (Ioo_subset_Ico_self hx)
+        (left_mem_Ico.mpr (lt_add_of_pos_right a hp.out))).mp hx')
+    map_target' := by
+      intro x hx
+      exact (eq_left_or_mem_Ioo_of_mem_Ico (equivIco p a x).2).resolve_left
+        (hx ∘ ((equivIco p a).symm_apply_apply x).symm.trans ∘ congrArg _)
+    left_inv' :=
+      fun x hx ↦ congrArg _ ((equivIco p a).apply_symm_apply ⟨x, Ioo_subset_Ico_self hx⟩)
+    right_inv' := fun x _ ↦ (equivIco p a).symm_apply_apply x
+    open_source := isOpen_Ioo
+    open_target := isOpen_compl_singleton
+    continuousOn_toFun := (AddCircle.continuous_mk' p).continuousOn
+    continuousOn_invFun := by
+      exact ContinuousAt.continuousOn
+        (fun _ ↦ continuousAt_subtype_val.comp ∘ continuousAt_equivIco p a) }
```

Comment:
Do you see this as being useful in its own right, or just a stepping-stone to `isLocalHomeomorph_coe`? If the former, it might be worth adding a couple of accompanying `rfl`-lemmas (so users don't have to immediately unfold the definition to prove anything about it).

Reply:
I've tagged the definition with `@simps` to add these `rfl`-lemmas.